osbuild/pipeline.py

```diff
@@ -44,14 +44,25 @@ def cleanup(*objs):
 
 
 class BuildResult:
-    def __init__(self, origin, returncode, output, error):
+    def __init__(self, origin: 'Stage', returncode: int, output: str, error: Dict[str, str]) -> None:
         self.name = origin.name
         self.id = origin.id
         self.success = returncode == 0
         self.output = output
         self.error = error
 
-    def as_dict(self):
+    def as_dict(self) -> Dict[str, Any]:
+        return vars(self)
```

Comment:
I really prefer doing this explicitly, so returning a new dict with the attributes we want to export.